src/io/mod.rs

```diff
@@ -0,0 +1,202 @@
+pub use file_reader::{LocalFsReaderConfig, ReaderConfiguration};
+
+mod file_reader {
```

Comment:
Could this file be renamed to file_reader.rs to avoid explicitly namespacing most of the file?

Reply:
Yes, it was temporal due to development. "file_writer" will happen in the future too so.